backend/api/db/crud/beneficiary.py

```diff
@@ -27,6 +34,112 @@
 )
 
 
+# see https://betakuang.medium.com/why-postgresqls-on-conflict-cannot-find-my-partial-unique-index-552327b85e1
+firstname_lastname_date_of_birth_unique_idx = (
+    "(LOWER(trim(firstname)), LOWER(trim(lastname)), date_of_birth, deployment_id)"
+)
+
+
+async def get_beneficiaries_like(
+    connection: Connection,
+    beneficiary: BeneficiaryImport,
+    deployment_id,
+) -> list[Beneficiary]:
+    matching_beneficiaries: list[Record] = await connection.fetch(
+        """
+SELECT *
+FROM beneficiary
+WHERE (lower(trim(firstname)) = $1 AND lower(trim(lastname)) = $2 AND date_of_birth = $3)
```

Comment:
il manque le deploymentId ici (pour éviter de matcher les homonymes de pulsieurs déploiements)
```suggestion
WHERE (lower(trim(firstname)) = $1 AND lower(trim(lastname)) = $2 AND date_of_birth = $3 AND deployment_id = $5 )
```